server/service/hosts.go

```diff
@@ -86,6 +90,15 @@ func listHostsEndpoint(ctx context.Context, request interface{}, svc fleet.Servi
 		}
 	}
 
+	var mdmSolution *fleet.AggregatedMDMSolutions
+	if req.Opts.MDMIDFilter != nil {
+		var err error
+		mdmSolution, err = svc.AggregatedMDMSolutions(ctx, req.Opts.TeamFilter, *req.Opts.MDMIDFilter)
```

Comment:
If the mdmID does not exist in the database, what does the endpoint return?

Reply:
It would return `nil` (and no error), so the response payload would not have the mdm part (and the `hosts` array would be empty).

Reply:
Ok, we should be sure the frontend dev is able to differentiate that response so they can display the appropriate state for case. It sounds like they'd be able to do that by checking if the mdm part is there or not. From an API design perspective, did we ever settle on a pattern when to omit empty vs. when to return explicit `null`, empty array, etc.? It might be easier for the frontend to check explicit `null`.

Reply:
Ah gotcha, good call, I'll check with Rachel and Luke how they want to handle that. It looks like the software object would return a 404 error if it is not found, so if only for consistency the MDM data should probably do that too.

Reply:
Ok after validation on Slack (https://fleetdm.slack.com/archives/C03C41L5YEL/p1660578586029799?thread_ts=1660311683.804929&cid=C03C41L5YEL), we'll go with the current approach to omit it if the mdm_id does not exist (it is different than software, but it makes sense in this case because the stats are only generated once an hour, so there could be no stats for an mdm_id and yet that mdm_id might still be valid).